streams/src/main/java/org/apache/kafka/streams/state/internals/Segments.java

```diff
@@ -167,7 +172,11 @@ private void cleanup(final long segmentId) {
             if (segment != null && segment.id <= oldestSegmentId) {
                 segments.remove(segmentEntry.getKey());
                 segment.close();
-                segment.destroy();
+                try {
+                    segment.destroy();
+                } catch (IOException e) {
+                    log.error("Error destroying segment id {}", segment.id, e);
```

Comment:
I did not find any `toString` in `RocksDBStore` so I added one in `Segment` (which can then expose the name, which is not something that `RocksDBStore` knows about).